### PR TITLE
Feature core #43 state observer

### PR DESCRIPTION
**Pull request by aaronplusone**
# What does this MR do?
* Updates tag values to OMAC Packml Implementation Guide v1.00
* Changes PackML Itemized Stats "count" field from int32 to float32 to allow accumulation on non-integer stats
* Adds services for interfacing more directly with PackML state machine (send command, send event, invoke state change, get state)
* Adds service for requesting state information from an external state controller, e.g. for when PackML is functioning as an observer (get state)

# Dependencies?
None

# Review Checklist
* [ ]  Message/Service definitions are clear and concise

@charlescc9, please review.

### CODE REVIEW

File: srv/GetState.srv [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Get State Service
+# Gets current state from state controller
+# Used when PackML is functioning as a state observer
+
+# Empty Request
+---
+int32 state        # Current PackML state
+int32 substate     # non-standard state within standard PackML state (0 if no substate)
```
Comment (charlescc9) [inline srv/GetState.srv:8]: non-blocking style: should it be "sub_state"?

File: msg/ItemizedStats.msg [hunk 2]
```diff
@@ -1,7 +1,7 @@
-# This message contains itemized statiistics for PackML statistics reporting
+# This message contains itemized statistics for PackML statistics reporting
 # Each item containds a unique ID that defines the "item".  
 # Example usage includes, error and warning events.
 
 int16 id                          # unique identifier (user defined)
-int32 count                       # number of occurances
+float32 count                     # total value of itemized stat
```
Comment (charlescc9) [inline msg/ItemizedStats.msg:6]: If count has changed meaning from total occurances of the stat to the total value of the stat, should the name change as well (i.e. to "total_value"?)

File: srv/InvokeStateChange.srv [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Invoke State Change
+# Updates PackML use the desired state as the "current" state
+# Note: This violates the state machine, and will update timers only.
+#   It will not advance the underlying FSM state. Future events may trigger abnormal behavior.
+
+string name       # Name of the desired state
+int32 state_enum  # Enumeration of the desired state (See State.msg)
```
Comment (charlescc9) [inline srv/InvokeStateChange.srv:7]: Does an enum need to be int32, or can we get by with int8?

File: srv/SendEvent.srv [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+int8 UNDEFINED = 0
+int8 SC = 1
+int8 HOLD = 2
+int8 UNHOLD = 3
+int8 SUSPEND = 4
+int8 UNSUSPEND = 5
+int8 RESET = 6
+int8 CLEAR = 7
+
+int32 event_id    # Event to send to the state machine
```
Comment (charlescc9) [inline srv/SendEvent.srv:19]: Same as above; should this be int32 or int8?

File: srv/InvokeStateChange.srv (hunk 3 above)
Comment (aaronplusone) [reply srv/InvokeStateChange.srv:7]: int32 was chosen to pass Simple Message semantics without type casting, but 32-bit is probably overkill.

File: srv/SendEvent.srv (hunk 4 above)
Comment (aaronplusone) [reply srv/SendEvent.srv:19]: Same rationale as above.

File: msg/ItemizedStats.msg (hunk 2 above)
Comment (aaronplusone) [reply msg/ItemizedStats.msg:6]: I wanted to do a least harm approach here, but you have a point. Existing applications could coerce an int to this field and operate normally, while changing the field name would require applications to update.

**Pull request merged**